BSB_LAN/BSB_LAN.ino

```diff
@@ -6564,49 +6566,62 @@ void loop() {
 
 
 #ifdef LOGGER
-#if defined(ESP32)
-  uint64_t freespace = 0;
-  freespace = SD.totalBytes() - SD.usedBytes();
-#else
-  uint32_t freespace = 0;
-  freespace = SD.vol()->freeClusterCount();
-#endif
-  if ((LoggingMode & CF_LOGMODE_SD_CARD) && freespace > MINIMUM_FREE_SPACE_ON_SD) {
+  if ((LoggingMode & (CF_LOGMODE_SD_CARD|CF_LOGMODE_UDP))) {
     if (((millis() - lastLogTime >= (log_interval * 1000)) && log_interval > 0) || log_now > 0) {
-//    SetDateTime(); // receive inital date/time from heating system
       log_now = 0;
-      File dataFile = SD.open(datalogFileName, FILE_APPEND);
-
-      if (dataFile) {
-        for (int i=0; i < numLogValues; i++) {
-          int outBufLen = 0;
-          if (log_parameters[i] > 0) {
-            outBufLen += sprintf_P(outBuf + outBufLen, PSTR("%lu;%s;%d;"), millis(), GetDateTime(outBuf + outBufLen + 80), log_parameters[i]);
-#ifdef AVERAGES
+      File dataFile;
+      if (LoggingMode & CF_LOGMODE_SD_CARD) {
+        #if defined(ESP32)
+          uint64_t freespace = SD.totalBytes() - SD.usedBytes();
+        #else
+          uint32_t freespace = SD.vol()->freeClusterCount();
+        #endif
+        if (freespace > MINIMUM_FREE_SPACE_ON_SD) {
+          dataFile = SD.open(datalogFileName, FILE_APPEND);
+          if (!dataFile) {
+            // if the file isn't open, pop up an error:
+            printToWebClient(PSTR(MENU_TEXT_DTO "\r\n"));
+            printFmtToDebug(PSTR("Error opening %s!\r\n"), datalogFileName);
+          }
+        }
+      }
+      IPAddress broadcast_ip;
+      if (LoggingMode & CF_LOGMODE_UDP) {
+        #ifdef WIFI
+          IPAddress local_ip = WiFi.localIP();
+        #else
+          IPAddress local_ip = Ethernet.localIP();
+        #endif
+        broadcast_ip = IPAddress(local_ip[0], local_ip[1], local_ip[2], 0xFF);
```

Comment:
(this is an example why indention of preprocessor directives has to be separated. Because now it looks like as if broadcast_ip... and IPAddress local_ip... are not in the same code level)

Reply:
Point taken. :)